### PR TITLE
[Single Machine Performance] Send nightly builds to SMP ECR

### PR DESCRIPTION
**Pull request by blt**
### What does this PR do?

This commit introduces a new nightly job that will publish Agent amd64-a7 to Single Machine Performance's ECR, in preparation for a new nightly-only tool to be introduced in a follow-up PR. We have added `COMMIT_TIME` to the publish job definition to allow us to order imagines in time.

### Motivation

We intend to introduce a new nightly-only tool, see [this page](https://datadoghq.atlassian.net/wiki/spaces/SMP/pages/3092744360/Agent+Integration+RFC) for details. 

REF SMP-673

### CODE REVIEW

**Review by goxberry — APPROVED** [review]

File: .gitlab/dev_container_deploy/docker_linux.yml [hunk 1]
```diff
@@ -222,6 +222,19 @@ dev_nightly-a7:
       - IMG_SOURCES: ${SRC_AGENT}:v${CI_PIPELINE_ID}-${CI_COMMIT_SHORT_SHA}-7-jmx-amd64,${SRC_AGENT}:v${CI_PIPELINE_ID}-${CI_COMMIT_SHORT_SHA}-7-jmx-arm64
         IMG_DESTINATIONS: agent-dev:nightly-${CI_COMMIT_REF_SLUG}-${CI_COMMIT_SHORT_SHA}-py3-jmx,agent-dev:nightly-${CI_COMMIT_REF_SLUG}-py3-jmx
 
+# deploy nightly build to single-machine-performance
+single_machine_performance-nightly-amd64-a7:
+  extends: .docker_publish_job_definition
+  stage: container_build
```
Comment (spencergilbert) [inline .gitlab/dev_container_deploy/docker_linux.yml:228]: **❓ question:** ‏ Any reason for this particular `stage`, the jobs around it have `dev_container_deploy`, and the other jobs I see in `container_build` don't seem related.

```suggestion
  stage: dev_container_deploy
```

**Review by spencergilbert — APPROVED** [review]
LGTM

**Comment by blt** [comment]
I'll wait to merge this until I get a sign-off from someone on @DataDog/container-integrations.

**Review by AliDatadog — APPROVED** [review]

File: .gitlab/docker_common/publish_job_templates.yml [hunk 2]
```diff
@@ -11,6 +11,8 @@
     <<: *docker_variables
     IMG_VARIABLES: ""
     IMG_SIGNING: ""
+  before_script:
+    - export COMMIT_TIME=$(git show -s --format=%ct $CI_COMMIT_SHA)
```
Comment (clamoriniere) [inline .gitlab/docker_common/publish_job_templates.yml:15]: does the `COMMIT_SHA` is not enough?

**Review by clamoriniere — APPROVED** [review]

File: .gitlab/docker_common/publish_job_templates.yml (hunk 2 above)
Comment (goxberry) [reply .gitlab/docker_common/publish_job_templates.yml:15]: The motivation for including the commit time is to make it easier for viewers of Single Machine Performance's outputs -- which will eventually include Datadog notebooks and dashboards -- to see how performance has changed over time, ideally at a glance. This option is one of the lower development cost efforts for getting at that time information for visualization purposes.
